tests/utils/test_common.py

```diff
@@ -159,38 +178,123 @@ def test_cli_missing_log_level_default_used():
 
 
 @mock.patch.dict(os.environ, {"LOG_LEVEL": "WARNING"})
 def test_cli_log_level_overrides_env_var_used():
     """Test CLI log level has precedence over env var."""
     configure_logging_and_warnings(logging.DEBUG)
     rasa_logger = logging.getLogger("rasa")
-    rasa_logger.level == logging.DEBUG
+    assert rasa_logger.level == logging.DEBUG
     matplotlib_logger = logging.getLogger("matplotlib")
     # Default log level for libraries is currently ERROR
-    matplotlib_logger.level == logging.ERROR
+    assert matplotlib_logger.level == logging.ERROR
 
 
 @mock.patch.dict(os.environ, {"LOG_LEVEL": "WARNING", "LOG_LEVEL_MATPLOTLIB": "INFO"})
 def test_cli_missing_log_level_env_var_used():
     """Test CLI without log level uses env var for both rasa and libraries."""
     configure_logging_and_warnings()
     rasa_logger = logging.getLogger("rasa")
-    rasa_logger.level == logging.WARNING
+    assert rasa_logger.level == logging.WARNING
     matplotlib_logger = logging.getLogger("matplotlib")
-    matplotlib_logger.level == logging.INFO
+    assert matplotlib_logger.level == logging.INFO
+
+
+def test_cli_valid_logging_configuration() -> None:
+    logging_config_file = "data/test_logging_config_files/test_logging_config.yml"
+    configure_logging_from_file(logging_config_file=logging_config_file)
+    rasa_logger = logging.getLogger("rasa")
+
+    handlers = rasa_logger.handlers
+    assert len(handlers) == 1
+    assert isinstance(handlers[0], logging.FileHandler)
+    assert "test_handler" == rasa_logger.handlers[0].name
+
+    logging_message = "This is a test info log."
+    rasa_logger.info(logging_message)
+
+    handler_filename = handlers[0].baseFilename
+    assert Path(handler_filename).exists()
+
+    with open(handler_filename, "r") as logs:
+        data = logs.readlines()
+        logs_dict = json.loads(data[-1])
+        assert logs_dict.get("message") == logging_message
+
+        for key in ["time", "name", "levelname"]:
+            assert key in logs_dict.keys()
+
+    # reset logging config
+    reset_logging()
+
+
+@pytest.mark.parametrize(
+    "logging_config_file",
+    [
+        "data/test_logging_config_files/test_missing_required_key_invalid_config.yml",
+        "data/test_logging_config_files/test_invalid_value_for_level_in_config.yml",
+        "data/test_logging_config_files/test_invalid_handler_key_in_config.yml",
+    ],
+)
+def test_cli_invalid_logging_configuration(
+    logging_config_file: Text, caplog: LogCaptureFixture
+) -> None:
+    with caplog.at_level(logging.DEBUG):
+        configure_logging_from_file(logging_config_file=logging_config_file)
+
+    assert (
+        f"The logging config file {logging_config_file} could not be applied "
+        f"because it failed validation against the built-in Python "
+        f"logging schema." in caplog.text
+    )
+
+
+@pytest.mark.skipif(
+    sys.version_info.minor == 7, reason="no error is raised with python 3.7"
+)
```

Comment:
this is odd, is this documented?

Reply:
what is the behaviour? Is logging set up properly or no?

Reply:
Yes, logging is set up properly since the invalid configs fail in Python 3.8 with ValueError (that's what I originally had installed in local dev environment). When I switched to 3.7, this particular invalid config doesn't raise any error in the logging module (as it does in 3.8); it seems in that Python release, the invalid config that I am testing is somehow valid. I didn't look too much in depth in the python 3.7 code and compare it to 3.8 to see where the change was made.

Reply:
taking a guess, it could be because of [the introduction of `validate` parameter since 3.8](https://docs.python.org/3/library/logging.config.html?highlight=dictconfig#dictionary-schema-details). I think it's ok to skip